docbook/Developers_Guide/en-US/Events_Reference_Bug.xml

```diff
@@ -277,6 +297,102 @@
 			</blockquote>
 		</blockquote>
 
+		<blockquote id="dev.eventref.bug.action.reportbugpre">
+			<title>EVENT_BUG_CREATE_PRE (Execute)</title>
+
+			<blockquote>
+				<para>
+					This event allows plugins to perform pre-processing of a new bug data
+					structure before the data is saved to the database. At this point,
+					the issue ID is not yet known, as the data has not yet been persisted.
+					Modifications on this structure can be made in the BugData object
+					received as parameter.
+					This event is triggered by the BugData API by any action that uses
+					the create() method.
+				</para>
+
+				<para>
+					Note: As the create action may be altered or stopped by other plugins,
+					or a final validation step before saving the data, the actions implemented
+					by this event should not have side effects outside the BugData structure.
+					Use BugData callbacks to be executed after the data is committed, and before
```

Comment:
I would rather if plugins don't register callbacks, but depend on PRE vs. POST events to do the logic without side-effects in PRE and do the necessary actions in POST.  We could also consider having an event type where if a plugin fail, we would stop the execution  of the rest (for PRE events), and another where we would continue with other plugins (for POST events).